ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/InsertPassword.hs

```diff
@@ -0,0 +1,153 @@
+
+import qualified Graphics.UI.Qtah.Core.QObject as QObject
+import qualified Graphics.UI.Qtah.Gui.QIcon as QIcon
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QCheckBox as QCheckBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLayout as QLayout
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+
+data InsertPassword = InsertPassword
+    { insertPassword :: QDialog.QDialog
+    , hasPassword :: QCheckBox.QCheckBox
+    , password :: QLineEdit.QLineEdit
+    , passwordWidget :: QWidget.QWidget
+    , confirmButton :: QPushButton.QPushButton
+    }
+
+data InsertPasswordResult
+    = InsertPasswordCanceled
+    | InsertPasswordAccepted (Maybe Text)
+
+makeLensesWith postfixLFields ''InsertPassword
+
+createPasswordField :: Text -> IO (QHBoxLayout.QHBoxLayout, QLineEdit.QLineEdit)
+createPasswordField placeholder = do
+    layout <- QHBoxLayout.new
+    field <- QLineEdit.new
+    QLineEdit.setEchoMode field QLineEdit.Password
+    QLineEdit.setPlaceholderText field $ toString placeholder
+
+    visibleButton <- QPushButton.new
+    void $ setProperty visibleButton ("styleRole" :: Text) ("passwordVisibilityToggle" :: Text)
+    QAbstractButton.setIcon visibleButton =<< QIcon.newWithFile (":/images/hide-pass-ic.png" :: String)
+    QAbstractButton.setCheckable visibleButton True
+    QWidget.setSizePolicyRaw visibleButton Maximum Maximum
+
+    QBoxLayout.addWidget layout field
+    QBoxLayout.addWidget layout visibleButton
+    QLayout.setSpacing layout 12
+    QBoxLayout.setStretch layout 0 1
+    QBoxLayout.setStretch layout 1 0
+
+    let
+      togglePasswordVisibility checked = QLineEdit.setEchoMode field $
+          if checked then QLineEdit.Normal else QLineEdit.Password
+
+    connect_ visibleButton QAbstractButton.toggledSignal togglePasswordVisibility
+
+    return (layout, field)
+
+initInsertPassword :: IO InsertPassword
+initInsertPassword = do
+    insertPassword <- QDialog.new
+    QObject.setObjectName insertPassword ("insertPasswordDialog" :: String)
+    QWidget.setWindowTitle insertPassword ("Insert Password" :: String)
+    QWidget.adjustSize insertPassword
+
+    layout <- createLayout insertPassword
+
+    hasPasswordLabel <- QLabel.newWithText $ toString passwordLabelText
+    QLabel.setWordWrap hasPasswordLabel True
+    QWidget.setMinimumSizeRaw hasPasswordLabel 600 30
+    hasPassword <- QCheckBox.new
```

Comment:
Can you explain why is this needed here? I think it's kinda silly when password manager requests a password when a wallet does not have one.

Also, I was unable to create new address in paswordless wallet - a dialog runs, I de-check this checkbox, click confirm and then get `Address generation failed due to incorrect passphrase`.